tests/conftest.py

```diff
@@ -713,3 +713,77 @@ def rook_repo(request):
     get_rook_repo(
         config.RUN['rook_branch'], config.RUN.get('rook_to_checkout')
     )
+
+
+@pytest.fixture(scope="session")
+def memory_leak_function(request):
+    """
+    Memory leak thread part which will be executed parallel with test run
+    Memory leak data will be captured in all worker nodes for ceph-osd process
+    Data will be appended in /tmp/(worker)-top-output.txt file for each worker
+    During teardown created tmp files will be deleted
+    """
+    temp_file = tempfile.NamedTemporaryFile(
+        mode='w+', prefix='test_status', delete=False
+    )
+
+    def get_flag_status():
+        with open(temp_file.name, 'r') as t_file:
+            return t_file.readline()
+
+    def set_flag_status(value):
+        with open(temp_file.name, 'w') as t_file:
+            t_file.writelines(value)
+
+    set_flag_status('running')
+
+    def finalizer():
+        """
+        Stop memory leak data capture thread and cleanup the files
+        """
+        set_flag_status('terminated')
+        try:
+            for status in TimeoutSampler(90, 3, get_flag_status):
+                if status == 'terminated':
+                    break
+        except TimeoutExpiredError:
+            log.warning(
+                "Background test execution still in progress before"
+                "memory leak thread terminated"
+            )
+        if thread:
+            thread.join()
+        for worker in helpers.get_worker_nodes():
+            if os.path.exists(f"/tmp/{worker}-top-output.txt"):
+                os.remove(f"/tmp/{worker}-top-output.txt")
+        log.info(f"Memory leak capture has stopped")
+
+    request.addfinalizer(finalizer)
```

Comment:
call this line before setting the flag in line 738

Reply:
finalizer is not getting identified if am using before line 738.

Reply:
Done, resolving this comment